Simulator/game_round.py

```diff
@@ -87,101 +89,71 @@ def __init__(self, game_players, pool_obj, step_func_obj):
-                # Behavior is to fight a random player.
-                elif len(player_nums) == 1 or players_matched == config.NUM_PLAYERS - 1 - self.NUM_DEAD:
-                    player_index = random.randint(0, len(players) - 1)
-                    while ((not players[player_index]) or players[num].opponent == players[player_index]
-                            or num == player_index):
-                        player_index = random.randint(0, len(players) - 1)
-                    player_copy = player_class(self.pool_obj, players[player_index].player_num)
-                    player_copy.board = players[player_index].board
-                    player_copy.chosen = players[player_index].chosen
-                    players[num].start_time = time.time_ns()
-                    config.WARLORD_WINS['blue'] = players[num].win_streak
-                    config.WARLORD_WINS['red'] = player_copy.win_streak
-                    index_won, damage = champion.run(champion.champion, players[num], player_copy,
-                                                     self.ROUND_DAMAGE[round_index][1])
-                    # if the alive player loses to a dead player, the dead player's reward is
-                    # given out to all other alive players
-                    alive = []
-                    for other in players:
-                        if other:
-                            if other.health > 0 and other is not players[num]:
-                                alive.append(other)
-                    if index_won == 2 or index_won == 0:
-                        players[num].health -= damage
-                        players[num].loss_round(player_round)
-                        if len(alive) > 0:
-                            for other in alive:
-                                other.won_ghost(damage/len(alive))
-                    players[num].combat = True
-                    players_matched += 1
-                else:
-                    return False
+        for match in self.matchups:
+            if not match[1] == "ghost":
+                # Assigning a battle
+                if not players[match[0]] or not players[match[1]]:
+                    print(match[0], match[1])
+                players[match[0]].opponent = players[match[1]]
+                players[match[1]].opponent = players[match[0]]
+                # Fixing the time signature to see how long battles take.
+                players[match[0]].start_time = time.time_ns()
+                players[match[1]].start_time = time.time_ns()
+                config.WARLORD_WINS['blue'] = players[match[0]].win_streak
+                config.WARLORD_WINS['red'] = players[match[1]].win_streak
+
+                # Main simulation call
+                index_won, damage = champion.run(champion.champion, players[match[0]], players[match[1]],
+                                                 self.ROUND_DAMAGE[round_index][1])
+
+                # Draw
+                if index_won == 0:

```

Comment:
Create a new method for drawn round. This should either give 0 reward or a positive reward should be given to all players that are not included in this fight.

Reply:
Also, it shouldn't count towards win streaks or loss streaks.